server/src/main/java/org/elasticsearch/search/aggregations/support/CoreValuesSourceType.java

```diff
@@ -240,28 +244,21 @@ public ValuesSource getScript(AggregationScript.LeafFactory script, ValueType sc
         }
 
         @Override
-        public ValuesSource getField(FieldContext fieldContext, AggregationScript.LeafFactory script) {
-            ValuesSource.Numeric dataSource = fieldData(fieldContext);
+        public ValuesSource getField(FieldContext fieldContext, AggregationScript.LeafFactory script, AggregationContext context) {
+            ValuesSource.Numeric dataSource = fieldData(fieldContext, context);
             if (script != null) {
                 // Value script case
                 return new ValuesSource.Numeric.WithScript(dataSource, script);
             }
             return dataSource;
         }
 
-        private ValuesSource.Numeric fieldData(FieldContext fieldContext) {
+        private ValuesSource.Numeric fieldData(FieldContext fieldContext, AggregationContext context) {
             if ((fieldContext.indexFieldData() instanceof IndexNumericFieldData) == false) {
                 throw new IllegalArgumentException("Expected numeric type on field [" + fieldContext.field() +
                     "], but got [" + fieldContext.fieldType().typeName() + "]");
             }
-            if (fieldContext.fieldType().isSearchable() == false
-                    || fieldContext.fieldType() instanceof DateFieldType == false) {
-                /*
-                 * We can't implement roundingPreparer in these cases because
-                 * we can't look up the min and max date without both the
-                 * search index (the first test) and the resolution which is
-                 * on the DateFieldType.
-                 */
+            if (fieldContext.fieldType() instanceof DateFieldType == false) {
```

Comment:
This means that runtime fields can't do it. It makes me think that we're doing *something* wrong, but I think that is something to solve in a follow up.

Reply:
Agreed.  The `instanceof` check definitely smells wrong here, but I don't know what the right answer is.